compiler/src/iree/compiler/Dialect/HAL/Target/WebGPU/test/is_nan.mlir

```diff
@@ -0,0 +1,58 @@
+// RUN: iree-opt --split-input-file --iree-hal-transformation-pipeline \
+// RUN:   --verify-diagnostics %s
+
+// RUN: iree-opt --split-input-file --iree-hal-transformation-pipeline \
+// RUN:   --verify-diagnostics %s 2>&1 | FileCheck %s
+
+// This test generates the 'isNan' function, which fails to compile.
+// TODO(#10142): Fix compilation and test success (fold into smoketest.mlir?)
+//     (or delete - operation coverage should probably use tests/e2e/ instead)
```

Comment:
@antiagainst we discussed this a bit - I'm expecting these tests to be changed or removed as we figure out the best path forward. For now, the tests show current behavior.

Reply:
SG!

Reply:
Why is it useful to have a test that enforces failure here? XFAIL makes sense to me when you've got a big suite and want to know if something suddenly starts working, but seems odd here for a single failure that's being actively worked on.

Reply:
Only trivial programs are working right now. Each nontrivial program that I've tried has ran into the issues in these tests. I narrowed down the issues to these minimal reproducers, so we can poke that them easily. Once these tests pass, I'm expecting to replace them with a simple smoketest and then enable the full coverage from tests/e2e/...